host/vm-manager/0001-Add-support-for-dual-display-in-vm-manager-for-Andro.patch

```diff
@@ -0,0 +1,143 @@
+From f4de1463e8563c818410b7a89cf9163203aa79b3 Mon Sep 17 00:00:00 2001
+From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
+Date: Wed, 19 Apr 2023 16:20:24 +0530
+Subject: [PATCH] Add support for dual display in vm-manager for Android.
+
+---
+ src/guest/guest.c         |  1 +
+ src/guest/start.c         | 62 +++++++++++++++++++++++++++++++++++----
+ src/include/guest/guest.h |  5 ++++
+ 3 files changed, 62 insertions(+), 6 deletions(-)
+
+diff --git a/src/guest/guest.c b/src/guest/guest.c
+index 4d2bcf4..d9a24ac 100644
+--- a/src/guest/guest.c
++++ b/src/guest/guest.c
+@@ -35,6 +35,7 @@ keyfile_group_t g_group[] = {
+ 	{ "mediation", { "battery_med", "thermal_med", "camera_med",NULL }},
+ 	{ "aaf",      { "path", "support_suspend", "audio_type", NULL } },
+ 	{ "guest_control", { "time_keep", "pm_control", NULL }},
++	{ "display",  { "connectors.0", "connectors.1", "show_fps", NULL } },
+ 	{ "extra",    { "cmd", "service", NULL } },
+ };
+ 
+diff --git a/src/guest/start.c b/src/guest/start.c
+index 25e1207..c4d6371 100644
+--- a/src/guest/start.c
++++ b/src/guest/start.c
+@@ -26,6 +26,7 @@
+ #include <regex.h>
+ #include <string.h>
+ #include <ctype.h>
++#include <menu.h>
+ #include "vm_manager.h"
+ #include "guest.h"
+ #include "utils.h"
+@@ -535,14 +536,63 @@ static int setup_sriov(GKeyFile *gkf, char **p, size_t *size)
+ 	int monitor_id = -1;
+ 	int vf = 0;
+ 	int mem = 0;
++	int number_of_displays = 0;
++	g_autofree gchar *connectors_0 = NULL;
++	g_autofree gchar *connectors_1 = NULL;
++	bool connector_0_valid = false;
++	bool connector_1_valid = false;
++	g_autofree gchar *show_fps = "";
++	char show_fps_str[15] = "show-fps=";
++	char connectors[8][8] = {
++		"HDMI-1",
++		"HDMI-2",
++		"HDMI-3",
++		"HDMI-4",
++		"DP-1",
++		"DP-2",
++		"DP-3",
++		"DP-4",
++	};
++	char connector_str[30] = {0};
++
+ 	GError *error = NULL;
+ 	keyfile_group_t *g = NULL;
+ 
+ 	g = &g_group[GROUP_VGPU];
+ 	monitor_id = g_key_file_get_integer(gkf, g->name, g->key[VGPU_MON_ID], &error);
++
++	g = &g_group[GROUP_DISPLAY];
++	show_fps = g_key_file_get_string(gkf, g->name, g->key[SHOW_FPS], &error);
++	if (show_fps != NULL)
++		strcat(show_fps_str, !strcmp(show_fps,"on") ? "on": "off");
```

Comment:
There is a problem with this logic.
If the user set show_fps=1, then it will silently convert it to off and add to qemu command.
The user will wonder what is wrong and submit a ticket because it can be misunderstood as faulty functionality.
If possible, please throw an error if the user provides a wrong input, and not silently correct the user.

Reply:
Updated the comment stating only "on" is valid for show_fps. Now FPS will be shown only if user sets "on". Rest of the values will result in off.

Reply:
No, this is wrong. Pass the string if only is either "on" or "off".
Otherwise just throw an error. Do not silently correct the input error and default to off.